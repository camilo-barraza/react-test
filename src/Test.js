import React, { Component } from 'react'
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux'
import { doTest, addTodo } from './actions'

class Test extends Component {

  constructor(props) {
    super(props)  
  }

  greet(){
    this.props.doTest('probando esa vaina')
  }

  render() {
    return ( <div>
      redux: {this.props.mono}
      <div className='test-field'>
        {this.props.otraVaina}
      </div>
      <div onClick={() => { this.props.addTodo('hey que ') }}> add todo </div>
      <div onClick={this.greet.bind(this)} > greet </div>
      <div className='clickable' onClick={() => { this.props.toggle() }}>clickable</div>
    </div>)
  }
}

const mapStateToProps = (state, ownProps) => ({
  mono: state.mono,
  otraVaina: ownProps.testing
})

const mapDispatchToProps = (dispatch, ownProps) => bindActionCreators({ doTest, addTodo }, dispatch)

export { Test }
export const TestRedux = connect(mapStateToProps, mapDispatchToProps)(Test)


// export default connect(mapStateToProps, mapDispatchToProps)(Test)