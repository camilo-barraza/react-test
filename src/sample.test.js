
import { Test } from "./Test";

it("renders correctly", () => {
  const wrapper = shallow(
    <Test mono='mono' otraVaina='otraVaina'/>
  );

  expect(wrapper).toMatchSnapshot();
});


it("formats temp correctly", () => {
  // mount our Temperature component
  const wrapper = mount(
    <Test mono='mono' otraVaina='otraVaina' />
  );

  // extract the text from the LargeText styled component
  const text = (wrapper.find("div.test-field")).text();
  console.log(text)
  // ensure it matches what is expected
  expect(text).toEqual("otraVainadf");
});

it("calls toggle on click", () => {
  // create a spy function
  const spy = sinon.spy();
  // pass spy function as our toggleForecast prop
  const wrapper = mount(
    <Test mono='mono' otraVaina='otraVaina' toggle={spy}/>
  );

  // find the first div and simulate a click event on it
  wrapper
    .find("div.clickable")
    .first()
    .simulate("click");

  // ensure that our spy (toggleForecast) was called when click was simulated
  expect(spy.calledOnce).toBe(true);
});