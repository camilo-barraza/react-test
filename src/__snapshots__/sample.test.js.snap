// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  redux: 
  mono
  <div>
    otraVaina
  </div>
  <div
    onClick={[Function]}
  >
     add todo 
  </div>
  <div
    onClick={[Function]}
  >
     greet 
  </div>
</div>
`;
